interfaces/core.go

```diff
@@ -25,56 +25,56 @@ import (
 	"regexp"
 )
 
-// Skill represents a capacity offered by a snap.
-type Skill struct {
-	Name  string
-	Snap  string
-	Type  string
-	Attrs map[string]interface{}
-	Apps  []string
-	Label string
+// Plug represents a capacity offered by a snap.
+type Plug struct {
+	Name      string
+	Snap      string
+	Interface string
+	Attrs     map[string]interface{}
+	Apps      []string
+	Label     string
 }
 
-// Slot represents the potential of a given snap to use a skill.
+// Slot represents the potential of a given snap to connect to a plug.
 type Slot struct {
-	Name  string
-	Snap  string
-	Type  string
-	Attrs map[string]interface{}
-	Apps  []string
-	Label string
+	Name      string
+	Snap      string
+	Interface string
+	Attrs     map[string]interface{}
+	Apps      []string
+	Label     string
 }
 
-// Type describes a group of interchangeable capabilities with common features.
-// Types are managed centrally and act as a contract between system builders,
+// Interface describes a group of interchangeable capabilities with common features.
+// Interfaces are managed centrally and act as a contract between system builders,
 // application developers and end users.
-type Type interface {
-	// Unique and public name of this type.
+type Interface interface {
+	// Unique and public name of this interface.
 	Name() string
 
-	// SanitizeSkill checks if a skill is correct, altering if necessary.
-	SanitizeSkill(skill *Skill) error
+	// SanitizePlug checks if a plug is correct, altering if necessary.
+	SanitizePlug(plug *Plug) error
 
 	// SanitizeSlot checks if a slot is correct, altering if necessary.
 	SanitizeSlot(slot *Slot) error
 
-	// SkillSecuritySnippet returns the configuration snippet needed by the
-	// given security system to allow a snap to offer a skill of this type.
+	// PlugSecuritySnippet returns the configuration snippet needed by the
+	// given security system to allow a snap to offer a plug of this interface.
 	//
-	// An empty snippet is returned when the skill doesn't require anything
+	// An empty snippet is returned when the plug doesn't require anything
 	// from the security system to work, in addition to the default
-	// configuration.  ErrUnknownSecurity is returned when the skill cannot
+	// configuration.  ErrUnknownSecurity is returned when the plug cannot
 	// deal with the requested security system.
-	SkillSecuritySnippet(skill *Skill, securitySystem SecuritySystem) ([]byte, error)
+	PlugSecuritySnippet(plug *Plug, securitySystem SecuritySystem) ([]byte, error)
```

Comment:
Unrelated to the rename, but this should also have the slot in. It's symmetrical to the respective slot method.